lib/Dialect/FIRRTL/Ops.cpp

```diff
@@ -74,6 +74,21 @@ static LogicalResult verifyCircuitOp(CircuitOp &circuit) {
     return failure();
   }
 
+  // Check that no extmodule defname conflicts with another module
+  // name in the circuit.
+  for (auto &op : *circuit.getBody()) {
+    if (auto extModule = dyn_cast<FExtModuleOp>(op)) {
+      auto defname = extModule.defnameAttr();
+      if (defname && circuit.lookupSymbol(defname.getValue()) &&
+          isa<FModuleOp>(circuit.lookupSymbol(defname.getValue()))) {
+        op.emitOpError(
+            "attribute 'defname' with value '" + defname.getValue() +
+            "' conflicts with the name of another module in the circuit");
```

Comment:
Please add a note pointing to the conflicting declaration, something like:

auto diag = op.emitOpError("stuff above);
diag.attachNote(otherOp.getLoc()) << "previous declaration here";

Reply:
Great suggestion. I'll add this.

Reply:
Added in a follow-up commit.